common/djangoapps/student/views.py

```diff
@@ -357,24 +357,29 @@ def signin_user(request):
     external_auth_response = external_auth_login(request)
     if external_auth_response is not None:
         return external_auth_response
+    # Determine the URL to redirect to following login:
+    redirect_to = get_next_url_for_login_page(request)
     if request.user.is_authenticated():
-        return redirect(reverse('dashboard'))
+        return redirect(redirect_to)
+
+    third_party_auth_error = None
+    for msg in messages.get_messages(request):
+        if msg.extra_tags.split()[0] == "social-auth":
+            third_party_auth_error = unicode(msg)
```

Comment:
@rlucioni I understand, but sometimes the user *does* want to see this message. For example, this is how the `NotActivatedException` (`_('This account has not yet been activated. An activation email has been re-sent.')`) gets passed on to the user. It could be frustrating to not know why the login didn't work without that explanation. In that case, it will be translated, since the exception message is defined in our code. Other messages may not be translated (since they're provided by external code), but they should hopefully be rare.